src/transformStateWithClones.js

```diff
@@ -1,13 +1,58 @@
 'use strict';
-
 /**
  * @param {Object} state
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const report = [];
+
+  for (const action of actions) {
+    const stateClone =
+      report.length === 0 ? { ...state } : { ...report[report.length - 1] };
```

Comment:
The cloning of the state is done correctly here using the spread operator. This ensures that each action operates on a separate copy of the state.